ldap/servers/slapd/back-ldbm/db-mdb/mdb_layer.c

```diff
@@ -2648,6 +2648,53 @@ dbmdb_get_entries_count(dbi_db_t *db, dbi_txn_t *txn, int *count)
     return dbmdb_map_error(__FUNCTION__, rc);
 }
 
+static int
+dbmdb_get_page_count(dbi_db_t *db, dbi_txn_t *txn, uint32_t *count)
+{
+    dbmdb_dbi_t *dbmdb_db = (dbmdb_dbi_t*)db;
+    MDB_stat stats = {0};
+    int rc = 0;
+
+    rc = START_TXN(&txn, txn, TXNFL_RDONLY);
+    rc = mdb_stat(TXN(txn), dbmdb_db->dbi, &stats);
+    if (rc == 0) {
+        *count = stats.ms_branch_pages + stats.ms_leaf_pages + stats.ms_overflow_pages;
+    }
+
+    END_TXN(&txn, 1);
+    return dbmdb_map_error(__FUNCTION__, rc);
+}
+
+/*
+ * Get the page count for this backend instance
+ * If any error occurs just return 0
+ */
+uint32_t
+dbmdb_get_inst_page_count(struct ldbminfo *li, ldbm_instance *inst)
+{
+    if (!inst->inst_id2entry) {
+        dbmdb_ctx_t *ctx = MDB_CONFIG(li);
+        int rc = 0;
+
+        if (!ctx->env) {
+            return 0;
```

Comment:
in bdb case, I think that you create the env, maybe we should do the same with lmdb if it is not already open ...